rest/route/middleware_test.go

```diff
@@ -475,6 +476,13 @@ func TestTaskAuthMiddleware(t *testing.T) {
 	m.ServeHTTP(rw, r, func(rw http.ResponseWriter, r *http.Request) {})
 	assert.NotEqual(http.StatusOK, rw.Code)
 
+	assert.NoError(task.UpdateOne(bson.M{"_id": "completedTask"}, bson.M{"$set": bson.M{"finish_time": time.Now().Add(-30 * time.Minute)}}))
+
+	r.Header.Set(evergreen.TaskHeader, "completedTask")
+	rw = httptest.NewRecorder()
+	m.ServeHTTP(rw, r, func(rw http.ResponseWriter, r *http.Request) {})
+	assert.Equal(http.StatusOK, rw.Code)
+
```

Comment:
The test above can pass with or without the auth change, so would be good to add a case where it's unauthorized because it's been a really long time.